applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_structural_response_function.cpp

```diff
@@ -46,227 +46,113 @@ namespace Kratos
     {
         KRATOS_TRY;
 
-        if (rResponseGradient.size() != rAdjointMatrix.size1())
-            rResponseGradient.resize(rAdjointMatrix.size1(), false);
+        if (rResponseGradient.size() != rResidualGradient.size1())
+            rResponseGradient.resize(rResidualGradient.size1(), false);
 
         rResponseGradient.clear();
 
         KRATOS_CATCH("");
     }
 
-    /// Calculate the local gradient w.r.t. second derivatives of primal solution.
-    /**
-     * @param[in]     rAdjointElem      the adjoint element.
-     * @param[in]     rAdjointMatrix    the transposed gradient of the
-     *                                  element's residual w.r.t. second derivatives.
-     * @param[out]    rResponseGradient the gradient of the response function.
-     * @param[in]     rProcessInfo      the current process info.
-     */
-    void AdjointStructuralResponseFunction::CalculateSecondDerivativesGradient(const Element& rAdjointElem,
-                                                    const Matrix& rAdjointMatrix,
-                                                    Vector& rResponseGradient,
-                                                    ProcessInfo& rProcessInfo)
+    void AdjointStructuralResponseFunction::CalculateSecondDerivativesGradient(const Element& rAdjointElement,
+                                   const Matrix& rResidualGradient,
+                                   Vector& rResponseGradient,
+                                   const ProcessInfo& rProcessInfo)
     {
         KRATOS_TRY;
 
-        if (rResponseGradient.size() != rAdjointMatrix.size1())
-            rResponseGradient.resize(rAdjointMatrix.size1(), false);
+        if (rResponseGradient.size() != rResidualGradient.size1())
+            rResponseGradient.resize(rResidualGradient.size1(), false);
 
         rResponseGradient.clear();
 
         KRATOS_CATCH("");
     }
 
     void AdjointStructuralResponseFunction::CalculateSecondDerivativesGradient(const Condition& rAdjointCondition,
-                                                    const Matrix& rAdjointMatrix,
-                                                    Vector& rResponseGradient,
-                                                    ProcessInfo& rProcessInfo)
+                                   const Matrix& rResidualGradient,
+                                   Vector& rResponseGradient,
+                                   const ProcessInfo& rProcessInfo)
     {
         KRATOS_TRY;
 
-        if (rResponseGradient.size() != rAdjointMatrix.size1())
-            rResponseGradient.resize(rAdjointMatrix.size1(), false);
+        if (rResponseGradient.size() != rResidualGradient.size1())
+            rResponseGradient.resize(rResidualGradient.size1(), false);
 
         rResponseGradient.clear();
 
         KRATOS_CATCH("");
     }
 
-
-    /// Calculate the scalar valued response function
     double AdjointStructuralResponseFunction::CalculateValue(ModelPart& rModelPart)
```

Comment:
Should probably rause an error instead of returning 0

Reply:
I agree.